lib/thrift/binary/framed/protocol_handler.ex

```diff
@@ -30,16 +30,30 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
 
     {recv_timeout, tcp_opts} = Keyword.pop(tcp_opts, :recv_timeout, @default_timeout)
 
-    transport_options = Keyword.put(tcp_opts, :packet, 4)
-    transport.setopts(socket, transport_options)
+    with {:ok, first_bytes} <- :gen_tcp.recv(socket, 4, recv_timeout),
```

Comment:
We receive 4 bytes but check only 1. Does it break if we only receive 1?

Reply:
Yes it breaks if we only receive 1. And it took me hours to debug why it wouldn't work.